<a href="https://colab.research.google.com/github/amislam2004/Milestone-Projects/blob/main/Islam_dj_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import plotly.express as px

In [ ]:
pip install pandas

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
'''
LOUNGER / QUIET TYPES AUDIENCE
'''
# Read CSV for the column you need for lounger audience
df = pd.read_csv("music_genre_classification.csv", usecols
= ['Artist Name', 'energy', 'loudness', 'acousticness'])

In [ ]:
# Get the smallest 10 energy values
low_energy = df.nsmallest(10, 'energy')
print(low_energy[['Artist Name', 'energy']])

                                            Artist Name   energy
6089                               Todd Anderson-Kunert  0.00002
2801         Ludwig van Beethoven, Emanuel Ax, Yo-Yo Ma  0.00433
1521  G. I. Gurdjieff, Anja Lechner, Vassilis Tsabro...  0.00526
7432                                              HILMA  0.00700
3561  Luigi Boccherini, Tim Hugh, Scottish Chamber O...  0.00714
1985                              Gerry Mulligan Sextet  0.00894
2919                                 Broccoli, you too?  0.00976
3373             Francisco Villegas, Julio Mariñelarena  0.01040
5833                                      Library Tapes  0.01080
3997                                        Carlos Cipa  0.01230


In [ ]:
# Plot a bar chart, using plotly
px.bar(low_energy, 'Artist Name', 'energy', title='10 Lowest Energy Songs by Artist',
color='energy', color_continuous_scale='Greens')


In [ ]:
# Get the lowest 10 loudness values in decibels
quietest = df.nsmallest(10, 'loudness')
print(quietest[['Artist Name', 'loudness']])
# Plot to a bar chart in plotly
px.bar(quietest, 'Artist Name', 'loudness', title='10 Quietest Songs by Artist', color='loudness', color_continuous_scale='blues')

                                            Artist Name  loudness
3561  Luigi Boccherini, Tim Hugh, Scottish Chamber O...   -36.544
2801         Ludwig van Beethoven, Emanuel Ax, Yo-Yo Ma   -32.639
1822  Alexander Glazunov, Alexander Rudin, Moscow Ph...   -32.520
3150                      Victor Kozlov, Irina Kulikova   -31.846
1268  Johann Sebastian Bach, Freiburger Barockorchester   -30.625
7432                                              HILMA   -29.955
1521  G. I. Gurdjieff, Anja Lechner, Vassilis Tsabro...   -29.845
6651              Ottorino Respighi, Magdalena Hoffmann   -29.234
3997                                        Carlos Cipa   -29.161
4895                                      Tony Anderson   -28.958


In [ ]:
# Get the highest 10 acousticness values
best_acoustics = df.nlargest(10, 'acousticness')
print(best_acoustics[['Artist Name', 'acousticness']])
# Plot to a bar chart in plotly
px.bar(best_acoustics, 'Artist Name', 'acousticness', title='10 Best Acoustic Songs by Artist', color='acousticness',color_continuous_scale='sunsetdark')

                           Artist Name  acousticness
757        Memphis Minnie & Kansas Joe         0.995
1264                  Blind Boy Fuller         0.995
3206                   Baby Boy Warren         0.995
5244  Anja Lechner, François Couturier         0.995
6349                     Raquel Meller         0.995
1404                  Daniel Raticelli         0.994
2124                     Juanita Reina         0.994
3649                   Thelonious Monk         0.994
4237              Antonio Garcia Isaac         0.994
6154                        R.J. Smith         0.994


In [ ]:
# Get the highest 30 acousticness values
best_acoustics = df.nlargest(30, 'acousticness')
print(best_acoustics[['Artist Name', 'acousticness']])
# Plot to a bar chart in plotly
px.bar(best_acoustics, 'Artist Name', 'acousticness', title='30 Best Acoustic Songs by Artist', color='acousticness',color_continuous_scale='sunsetdark')

                                      Artist Name  acousticness
757                   Memphis Minnie & Kansas Joe         0.995
1264                             Blind Boy Fuller         0.995
3206                              Baby Boy Warren         0.995
5244             Anja Lechner, François Couturier         0.995
6349                                Raquel Meller         0.995
1404                             Daniel Raticelli         0.994
2124                                Juanita Reina         0.994
3649                              Thelonious Monk         0.994
4237                         Antonio Garcia Isaac         0.994
6154                                   R.J. Smith         0.994
6426  Franz Schubert, Mischa Maisky, Daria Hovora         0.994
6700                                 Diane Edison         0.994
630                            Juanito Valderrama         0.993
688                               Lata Mangeshkar         0.993
893             Rekha, Vinod Khanna, Ram

In [ ]:
'''
Track team / partierS AUDIENCE
'''
# Read CSV for the column you need for lounger audience
df = pd.read_csv("music_genre_classification.csv", usecols
= ['Artist Name', 'energy', 'tempo', 'duration_in min/ms'])

In [ ]:
# Get the largestt 10 energy values
high_energy = df.nlargest(10, 'energy')
print(high_energy[['Artist Name', 'energy']])

            Artist Name  energy
370          Darkthrone   1.000
15            FEVER 333   0.998
4241          FEVER 333   0.998
6208               Lard   0.998
1387     Pretty Vicious   0.997
1650         Hellripper   0.997
4934         Hellhammer   0.997
5197  The Acacia Strain   0.997
283       Ozzy Osbourne   0.996
959        Funeral Mist   0.996


In [ ]:
# Plot a bar chart, using plotly
px.bar(high_energy, 'Artist Name', 'energy', title='10 Highest Energy Songs by Artist', color='energy', color_continuous_scale='purples')

In [ ]:
# Get the highest 10 energy values
high_energy = df.nlargest(10, 'energy')
print(high_energy[['Artist Name', 'energy']])

            Artist Name  energy
370          Darkthrone   1.000
15            FEVER 333   0.998
4241          FEVER 333   0.998
6208               Lard   0.998
1387     Pretty Vicious   0.997
1650         Hellripper   0.997
4934         Hellhammer   0.997
5197  The Acacia Strain   0.997
283       Ozzy Osbourne   0.996
959        Funeral Mist   0.996


In [ ]:
# Get the highest 10 loudness values in decibels
partiers = df.nlargest(10, 'tempo')
print(partiers[['Artist Name', 'tempo']])
# Plot to a bar chart in plotly
px.bar(partiers, 'Artist Name', 'tempo', title='10 High Tempo Songs by Artist', color='tempo', color_continuous_scale='icefire')

            Artist Name    tempo
3581                IDK  249.438
2579         New Candys  216.194
5150         New Candys  216.194
4424         frown line  216.091
7018  Angels & Airwaves  216.020
5465    Ólafur Arnalds  212.849
4385           AFN Peso  212.166
2390               Evil  209.657
94          Aisha Badru  208.648
3688      Johnny Winter  208.576


In [ ]:
# Get the short 10 duration_in min/ms values
short_duration = df.nsmallest(10, 'duration_in min/ms')
print(short_duration[['Artist Name', 'duration_in min/ms']])

                                Artist Name  duration_in min/ms
4010                        Mahendra Kapoor            0.687450
2132                           R. D. Burman            1.209367
4066             Ilaiyaraaja, K. J. Yesudas            1.291667
5974   Eleni Karaindrou, Alexandros Botinis            1.353333
6447                   Kavita Krishnamurthy            1.426883
725                           Reyanna Maria            1.429033
4274                           R. D. Burman            1.434450
609                            Bappi Lahiri            1.464217
5438  Mani Sharma, Amit Heri, Rajesh Vaidya            1.495367
7561                               Labrinth            1.531167


In [ ]:
# Plot to a bar chart in plotly
px.bar(short_duration, 'Artist Name', 'duration_in min/ms', title='10 short duration Songs by Artist', color='duration_in min/ms',color_continuous_scale='temps')

In [ ]:
# Get the short 10 duration_in min/ms values
short_duration = df.nsmallest(30, 'duration_in min/ms')
print(short_duration[['Artist Name', 'duration_in min/ms']])
# Plot to a bar chart in plotly
px.bar(short_duration, 'Artist Name', 'duration_in min/ms', title='30 short duration Songs by Artist', color='duration_in min/ms',color_continuous_scale='temps')

                                            Artist Name  duration_in min/ms
4010                                    Mahendra Kapoor            0.687450
2132                                       R. D. Burman            1.209367
4066                         Ilaiyaraaja, K. J. Yesudas            1.291667
5974               Eleni Karaindrou, Alexandros Botinis            1.353333
6447                               Kavita Krishnamurthy            1.426883
725                                       Reyanna Maria            1.429033
4274                                       R. D. Burman            1.434450
609                                        Bappi Lahiri            1.464217
5438              Mani Sharma, Amit Heri, Rajesh Vaidya            1.495367
7561                                           Labrinth            1.531167
2109                                     Applewood Road            1.608017
2475                                     PinkPantheress            1.642400
3635        

In [ ]:
'''
NYT bestseller book club / literary fans
'''
# Read CSV for the column you need for NYT best seller book club audience
df = pd.read_csv("music_genre_classification.csv", usecols
  = ['Artist Name', 'Popularity', 'danceability', 'speechiness'])

In [ ]:
# Get the highest 10 Popularity values
high_Popularity = df.nlargest(10, 'Popularity')
print(high_Popularity[['Artist Name', 'Popularity']])

                  Artist Name  Popularity
6062                Lil Nas X       100.0
3797            Justin Bieber        99.0
2359                 Doja Cat        98.0
1816              Masked Wolf        97.0
6736               Ed Sheeran        97.0
3315               Bruno Mars        96.0
4263                Bad Bunny        96.0
6038  J Balvin, Maria Becerra        95.0
777                The Weeknd        94.0
4190            Billie Eilish        93.0


In [ ]:
# Plot a bar chart, using plotly
px.bar(high_Popularity, 'Artist Name', 'Popularity', title='10 Highest Popularity Songs by Artist', color='Popularity', color_continuous_scale='darkmint')

In [ ]:
# Get the highest 10 Pouplarity values
high_Popularity = df.nlargest(10, 'Popularity')
print(high_Popularity[['Artist Name', 'Popularity']])

                  Artist Name  Popularity
6062                Lil Nas X       100.0
3797            Justin Bieber        99.0
2359                 Doja Cat        98.0
1816              Masked Wolf        97.0
6736               Ed Sheeran        97.0
3315               Bruno Mars        96.0
4263                Bad Bunny        96.0
6038  J Balvin, Maria Becerra        95.0
777                The Weeknd        94.0
4190            Billie Eilish        93.0


In [ ]:
# Get the literary fans values in decibels
literary_fans = df.nlargest(10, 'danceability')
print(literary_fans[['Artist Name', 'danceability']])
# Plot to a bar chart in plotly
px.bar(literary_fans, 'Artist Name', 'danceability', title='10 High Tempo Songs by Artist', color='danceability', color_continuous_scale='purples')

                   Artist Name  danceability
2326                Rebelution         0.975
4120                Gucci Mane         0.972
7116                      Tyga         0.968
3868               Jody Breeze         0.966
4930              Run–D.M.C.         0.965
4019  Ashnikko, Princess Nokia         0.964
4215                  Ashnikko         0.964
1576                 Boy 8-Bit         0.962
5417                Lou Hayter         0.962
6070                    John K         0.958


In [ ]:
# Get the high speechiness_in min/ms values
high_speechiness = df.nlargest(10, 'speechiness')
print(high_speechiness[['Artist Name', 'speechiness']])

                                            Artist Name  speechiness
1043                                       Ishmael Reed        0.960
5763                                       Amiri Baraka        0.938
7005  Rajesh Khanna, Sumita Sanyal, Ramesh Deo, Seem...        0.829
734                                    Amitabh Bachchan        0.814
4740                                    Langston Hughes        0.806
580                                     Margaret Walker        0.678
1602                                           Rob Frye        0.667
3646                                          Son House        0.658
4446                                       Moneybagg Yo        0.619
893                   Rekha, Vinod Khanna, Ram P. Sethi        0.565


In [ ]:
# Plot to a bar chart in plotly
px.bar(high_speechiness, 'Artist Name', 'speechiness', title='10 highest speechiness by Artist', color='speechiness',color_continuous_scale='twilight')

In [ ]:
# Get the high speechiness_in min/ms values
high_speechiness = df.nlargest(30, 'speechiness')
print(high_speechiness[['Artist Name', 'speechiness']])
# Plot to a bar chart in plotly
px.bar(high_speechiness, 'Artist Name', 'speechiness', title='30 highest speechiness by Artist', color='speechiness',color_continuous_scale='twilight')

                                            Artist Name  speechiness
1043                                       Ishmael Reed        0.960
5763                                       Amiri Baraka        0.938
7005  Rajesh Khanna, Sumita Sanyal, Ramesh Deo, Seem...        0.829
734                                    Amitabh Bachchan        0.814
4740                                    Langston Hughes        0.806
580                                     Margaret Walker        0.678
1602                                           Rob Frye        0.667
3646                                          Son House        0.658
4446                                       Moneybagg Yo        0.619
893                   Rekha, Vinod Khanna, Ram P. Sethi        0.565
6975                                            DijahSB        0.562
1962                                             Cordae        0.558
4864                                           Mr. Oizo        0.550
3744                              

12.	Run that cell and see your small table print and bar chart plot. What do you notice about this bar chart that’s different from the others? Why do you think it’s different?
This bar chart is percentage wise, and it is showing high to low acousticness dark purple to light yellow color respectively. 
13.	Now try changing the 10 value in the nlargest() function to 30, then run the cell again. Is it more clear why the bar chart was different? Explain.
Yes it is 

Create artist list and visualize for the other two audience types
Now that you’ve built bar charts for all of the traits for the lounger / quiet types, do this again for both the track team / partiers and the NYT bestseller book club / literary fans.

More Thought
Answer these remaining questions, as you build your final charts.
1.	Look closely at the track team charts for energy and tempo. There’s a duplicate in each. How does that interrupt your bar chart?
Duplicates indicate different songs with the same artist; for example, Fever 333 has a track “Bite Back” with the energy 0.998 and other track “Burn It” with the enery value 0.959. Also, track “Bite Back” is repeated just like in tempo coulumn, there is repeated track “Zyko” which needs to be cleaned. 
What should have been done with the data before plotting it to avoid this?
Going back to data and deleting the repeated rows can be helpful in that case.
2.	After you complete the bar charts for all three audience types, look for overlap. For example, what artists are present in all three charts for the lounger type? Write those down as must-have songs for your song list.
a	Loungers / quiet types
1	high acousticness
2	low energy
3	low loudness
There are overlaps and common artists for three subcategories, so they are must have tracks for the DJ:
Artist Name with low energy and quietest
2801         Ludwig van Beethoven, Emanuel Ax, Yo-Yo Ma  
1521  G. I. Gurdjieff, Anja Lechner, Vassilis Tsabro... 
7432                                              HILMA  
3561  Luigi Boccherini, Tim Hugh, Scottish Chamber O...  
3997                                        Carlos Cipa 

b	Track team / partiers
1	high energy
2	high tempo
3	short duration
No overlaps and no common artists for three subcategories.

c	NYT bestseller book club / literary fans
1	high popularity
2	high danceability
3	high speechiness
No common artists in literary fans.

